estimagic/optimization/reparametrize.py

```diff
@@ -420,3 +438,89 @@ def _fixed_from_internal(params_subset, constr):
     if value is not None:
         res["value"] = value
     return res["value"]
+
+
+def _rescale_to_internal(
```

Comment:
This should be broken up into two functions: One that calculates the scaling factor and one that does the rescaling based on the scaling factor. Then only the actual rescaling should be called in `reparametrize_to_internal`. The calculation of the scaling factor should be done in minimize and the resulting scaling factor should be part of the optimization output. 

My problem with the current implementation is that a call `reparametrize_to_internal` with a value that is different from the start values (for example if we want to calculate a gradient on the internal function), will trigger a re-calculation of the scaling factor. This not only slow but probably also not what the user wants. There should be at most one scaling factor per optimization.